### PR TITLE
Add dependencies versions requirement.

### PR DESCRIPTION
**Pull request by kokarez**
xtension-fftw can't support all version of xtensor so let's start the same system of versioning that is done by xtensor. The xtl/xtension version match the one on current master of xtension, and the fftw3 version match the last stable release. (which is available on Ubuntu)

### CODE REVIEW

**Comment by egpbos** [comment]
Thanks for trying to fix this and also #44 ! Once you think they are ready to merge, ping me so I can review it. I may not have time immediately, but I'll do it asap, probably on Monday (GMT+1 here).

**Comment by kokarez** [comment]
Not sure how you manage the release but is it possible to push a new release of xtensor - with the -fnoexception ? (on couda too).

**Comment by kokarez** [comment]
> Thanks for trying to fix this and also #44 ! Once you think they are ready to merge, ping me so I can review it. I may not have time immediately, but I'll do it asap, probably on Monday (GMT+1 here).

Ping :) (FYI I'm french) I decide to debug that until I felt asleep.

**Comment by egpbos** [comment]
> Not sure how you manage the release but is it possible to push a new release of xtensor - with the -fnoexception ? (on couda too).

@JohanMabille @SylvainCorlay is this possible? @kokarez what do you need it for?

**Comment by kokarez** [comment]
@kokarez what do you need it for?

We are converting some xtensor code to support no-exception. 

xtensor patch: https://github.com/xtensor-stack/xtensor/commit/ddf1d0bb021feeca632804adeac20942dc8927bf
xtl: https://github.com/xtensor-stack/xtl/pull/148

I will check on the xtensor side for the release :)

**Comment by kokarez** [comment]
The windows is failing due to: https://github.com/xtensor-stack/xtensor/issues/1878

**Comment by SylvainCorlay** [comment]
Please drop win-32 in the test matrix. We are not making conda-forge packages for win-32 anymore.

**Comment by kokarez** [comment]
Ok doing it rigth now

**Comment by kokarez** [comment]
Fixed!

File: .travis.yml [hunk 1]
```diff
@@ -144,32 +143,20 @@ install:
...
-          make install
-          cd $TRAVIS_BUILD_DIR
-        fi
+        conda install -y fftw -c conda-forge
       fi
     # Testing
     - mkdir $TRAVIS_BUILD_DIR/build
     - cd $TRAVIS_BUILD_DIR/build
-    - cmake -DBUILD_TESTS=ON -DCOVERAGE=$COVERAGE ..
+    - cmake -DBUILD_TESTS=ON -DDOWNLOAD_GTEST=ON -DCOVERAGE=$COVERAGE ..
```
Comment (SylvainCorlay) [inline .travis.yml]: No need to set BUILD_TESTS if DOWNLOAD_GTEST is set!

File: .travis.yml [hunk 2]
```diff
@@ -144,32 +143,20 @@ install:
       elif [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
         MINCONDA_OS=$MINCONDA_OSX;
       fi
-    - wget "http://repo.continuum.io/miniconda/Miniconda3-$MINCONDA_VERSION-$MINCONDA_OS.sh" -O miniconda.sh;
+    - wget --tries=10 "http://repo.continuum.io/miniconda/Miniconda3-$MINCONDA_VERSION-$MINCONDA_OS.sh" -O miniconda.sh;
     - bash miniconda.sh -b -p $HOME/miniconda
     - export PATH="$HOME/miniconda/bin:$PATH"
     - hash -r
     - conda update -yq conda
+    - conda install -y cmake xtl==0.6.7 xtensor=0.20.10 nlohmann_json=3.7.1 -c conda-forge
```
Comment (SylvainCorlay) [inline .travis.yml]: Why xtensor 0.20?

File: .travis.yml (hunk 1 above)
Comment (kokarez) [reply .travis.yml]: Did not try :) I will tho.

File: .travis.yml (hunk 2 above)
Comment (kokarez) [reply .travis.yml]: it was the release when I started this patch - but yea I just change to be the last one.

**Comment by kokarez** [comment]
fixed

**Comment by SylvainCorlay** [comment]
@egpbos I am not sure we wanted ^0.21 since it also works with ^0.20.10...

**Comment by egpbos** [comment]
Yes, let's keep the dependencies loose in Conda at least. In CMake it seems we cannot specify minimum dependencies. Probably the best way to do that in CMake is for the dependency package (xtensor in this case) to specify feature tags or something like that.

**Pull request merged**